src/sync/stream.ts

```diff
@@ -297,27 +442,81 @@ export class StreamManager extends AbortController {
     });
   }
 
-  addTx(tx: IHistoryTx) {
+  isQueueDone() {
+    // Must not be processing the queue and the queue must be empty
+    return !(this.isProcessingQueue && this.itemQueue.size() !== 0);
+  }
+
+  async processQueue() {
+    if (this.isProcessingQueue) {
+      return;
+    }
+    this.isProcessingQueue = true;
+    while (true) {
+      if (this.signal.aborted) {
+        // Abort processing the queue
+        break;
+      }
+      const item = this.itemQueue.dequeue();
+      if (this.itemQueue.size() <= MIN_QUEUE_SIZE_FOR_ACK && this.lastAcked <= this.lastProcSeq) {
+        if (!this.hasReceivedEndStream) {
+          // Send the ACK for the end of the queue
+          this.lastAcked = this.lastSeenSeq;
+          this.stats.ack(this.lastSeenSeq);
+          this.connection.sendStreamHistoryAck(this.streamId, this.lastSeenSeq);
+        }
+      }
+      if (!item) {
+        // Queue is empty
+        this.stats.queueEmpty();
+        break;
+      }
+      this.stats.proc();
+      this.lastProcSeq = item.seq;
+      if (isStreamItemAddress(item)) {
+        const addr = item.address;
+        const alreadyExists = await this.storage.isAddressMine(addr.base58);
+        if (!alreadyExists) {
+          await this.storage.saveAddress(addr);
+        }
+      } else if (isStreamItemVertex(item)) {
+        await this.storage.addTx(item.vertex);
+      }
```

Comment:
We enqueue items so having an unknown type would mean we put something wrong there.
The place we check fullnode messages is the listener (under `buildListener`) where we do log unknown events so we are only processing or acking addresses and vertices.

There is another issue of weather the vertex sent is actually a vertex but that would be a fullnode error since the "api contract" was broken.